app/src/main/java/com/sunasterisk/fooddaily/data/source/OnDataLoadedCallback.kt

```diff
@@ -0,0 +1,9 @@
+package com.sunasterisk.fooddaily.data.source
+
+import com.sunasterisk.fooddaily.data.model.FoodDetails
+import java.lang.Exception
+
+interface OnDataLoadedCallback {
+    fun onSuccess(data: List<FoodDetails>)
+    fun onFailure(exception: Exception)
+}
```

Comment:
Đặt tên rõ ràng hoặc là dùng generic type 
```suggestion
interface OnFoodDetailsLoadedCallback {
    fun onSuccess(data: List<FoodDetails>)
    fun onFailure(exception: Exception)
}
```

Reply:
Nên dùng Generic Type để tái sử dụng
```suggestion
interface OnDataLoadedCallback<T> {
    fun onSuccess(data: List<T>)
    fun onFailure(exception: Exception)
}
```

Reply:
vâng ạ